astropy/utils/exceptions.py

```diff
@@ -75,6 +75,18 @@ def __repr__(self):
 NoValue = _NoValue()
 
 
+def _add_note_to_exception(exc: Exception, note: str) -> Exception:
+    import sys
+
+    if sys.version_info >= (3, 11):
+        exc.add_note(note)
+        return exc
+    else:
+        # mimic Python 3.11 behavior:
+        # preserve error message and traceback
+        return type(exc)(f"{exc!s}\n{note}").with_traceback(exc.__traceback__)
```

Comment:
This is a side improvement but isn't strictly needed here, I just applied an existing snippet of mine here. Let me know if I should revert it or move it to a separate PR.

Reply:
Hum, I didn't think about that. I guess replacing a tuple attribute is objectively less evil than returning an entirely new instance in a context where we accept mutations anyway. Thanks !